### ISSUE TITLE
[hw/kmac] Data output hardening for output / internal state

### ISSUE THREAD

**Issue by moidx** [lowRISC/opentitan#6544]
Need to harden kmac's output path to make sure it is not a single point of failure to expose keymgr usage.

**Comment by tjaychen** [comment]
@eunchan @m-temp i think we might have missed this one. 
I think the area of interest is [here](https://github.com/lowRISC/opentitan/blob/master/hw/ip/kmac/rtl/kmac_app.sv#L615), can you confirm? 

If yes, we might want to...
1. make the `mux_sel` type a bit wider 
2. pass both the `mux_sel` and `reg_state_valid_o` signals through a [secure anchor buf](https://github.com/lowRISC/opentitan/blob/master/hw/ip/prim/rtl/prim_sec_anchor_buf.sv)

**Comment by m-temp** [comment]
Thx @tjaychen for pointing this out. When I had a brief overview a couple of weeks ago, I thought everything is fine here, because the sparse_fsm takes care of that `valid` signal.  However, I just noticed that `reg_state_valid_o` isn't used at all! See [kmac.sv#L957-L958](https://github.com/lowRISC/opentitan/blob/master/hw/ip/kmac/rtl/kmac.sv#L957-L958).

@eunchan Is there any reason for that? Shouldn't we at least AND the valid-Signal with the data or do I miss something here?

Anyways, I think both points should be implemented (However, a secure anchor buf for the `valid` signal makes only sense, if we use it...)

**Comment by tjaychen** [comment]
good point, i did not realize the `valid` isn't used either :) we should probably improve that.

**Comment by m-temp** [comment]
When hardening the mux, we should also add a check for `lc_ctrl_escalate_en` as suggested in #8982

**Comment by tjaychen** [comment]
i'm sorry I feel like I was looking at the wrong thing earlier.
Invalidating on escalation sounds good to me. 
Just to confirm, on escalation, you would probably mark the digest as not done, and that the digest does not assume the new value?

**Comment by m-temp** [comment]
Yes, I assigned you as a reviewer for #10744 There are only a couple of line of codes. I think looking at them is faster than me explaining it. Let me know if there any questions.

**Comment by m-temp** [comment]
#10744 is merged. can we close this, or is there anything missing/open from your side?

**Comment by tjaychen** [comment]
let's close this.
